![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)


<h2 align='center'>Data Literacy through Sports Analytics</h2>
<h3 align='center'>CATCA 2021</h3>

<h3 align='center'>Tina Leard (Cybera)<br>
Michael Lamoureux (University of Calgary)</h3><br>

<h4 align='center'> Slides at: https://tinyurl.com/callystosports</h4>

![Callysto.ca License](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-bottom.jpg?raw=true)

This is a slide show in Jupyter notebook format.<br>
The notebook is organized into a series of editable cells that can hold text, images, and code.

To start, select "Kernel" from the menu bar and choose "Restart & Run All".<br>
Confirm by clicking the "Restart and Run All Cells" button.

Return to the top of the notebook and click the first cell.

To start the slide show, click on the icon in the menu bar that looks like a bar chart, as shown here. 
<img src="images/slideshow_icon.png"
     alt="slide show icon"
     width=100 />
     
You can advance the slide show by using the space bar. Use Shift+SpaceBar to go backward.<br>
Exit by clicking on the "X" in the top left corner of the slide show. 

Note that the slide show is a live Jupyter notebook, so you can run the code and edit while in the slide show. 

<center><img src='./images/ccby.png' alt="CC BY logo" width='300' /></center>

<p><center><a href='https://creativecommons.org/licenses/by/4.0/'>CC BY</a>:<br>
This license allows reusers to distribute, remix, adapt, and build upon the material in any medium or format,<br>
so long as attribution is given to the creator. 
</center></p>

In [1]:
import numpy as np
import pandas as pd
from pandas import read_csv
import plotly.graph_objects as go
import plotly.express as px
from plotly import subplots
from plotly.offline import download_plotlyjs, plot,iplot
import cufflinks as cf
cf.go_offline()

from IPython.display import YouTubeVideo

from ipysheet import sheet, cell, cell_range

%matplotlib inline

# Overview

- Data literacy via sports
- The learning progression
- Examples of learning and data analysis
 - Professional soccer
 - Ice hockey
 - Field hockey
- Python, Jupyter, and Callysto

<center><img src='./images/data_literacy.png' alt='data literacy' width='85%' /></center>

## Example: professional soccer event data
Data source: https://github.com/metrica-sports/sample-data

In [2]:
df_soccer = pd.read_csv("data/Sample_Game_1_RawEventsData.csv"); df_soccer

Team       Type       Subtype  Period  Start Frame  Start Time [s]  \
0     Away  SET PIECE      KICK OFF       1            1            0.04   
1     Away       PASS           NaN       1            1            0.04   
2     Away       PASS           NaN       1            3            0.12   
3     Away       PASS           NaN       1           45            1.80   
4     Away       PASS           NaN       1           77            3.08   
...    ...        ...           ...     ...          ...             ...   
1740  Home       PASS           NaN       2       143361         5734.44   
1741  Home       PASS           NaN       2       143578         5743.12   
1742  Home  BALL LOST  INTERCEPTION       2       143598         5743.92   
1743  Away   RECOVERY       BLOCKED       2       143617         5744.68   
1744  Away   BALL OUT           NaN       2       143622         5744.88   

      End Frame  End Time [s]      From        To  Start X  Start Y  End X  \
0             0          0.00  Player19       NaN      NaN      NaN    NaN   
1             3          0.12  Player19  Player21     0.45     0.39   0.55   
2            17          0.68  Player21  Player15     0.55     0.43   0.58   
3            61          2.44  Player15  Player19     0.55     0.19   0.45   
4            96          3.84  Player19  Player21     0.45     0.32   0.49   
...         ...           ...       ...       ...      ...      ...    ...   
1740     143483       5739.32  Player12  Player13     0.60     0.33   0.19   
1741     143593       5743.72  Player13   Player4     0.09     0.88   0.14   
1742     143618       5744.72   Player4       NaN     0.13     0.69   0.07   
1743     143617       5744.68  Player16       NaN     0.05     0.62    NaN   
1744     143630       5745.20  Player16       NaN     0.05     0.63   0.03   

      End Y  
0       NaN  
1      0.43  
2      0.21  
3      0.31  
4      0.47  
...     ...  
1740   0.95  
1741   0.69  
1742   0.61  
1743    NaN  
1744   1.01  

[1745 rows x 14 columns]

**Home team passes, second half**

In [3]:
df_soccer.loc[lambda df: (df['Team'] == 'Home') & (df['Period'] == 2) & (df['Type'] == 'PASS'), :] \
   .iplot(kind="scatter",x = "Start X", y = "Start Y", mode = "markers")

## Bridging expert to novice

## Data visualization learning progression
<img src='./images/creating_scaffolding.png' alt='scaffolding' width='95%' />

## Data visualization learning progression
<img src='./images/creating_adapting.png' alt='adapting' width='95%' />

## Data gathering learning progression
<center><img src='./images/gathering.png' alt='gathering' width='85%' /></center>
Source: <a href='http://oceansofdata.org/sites/oceansofdata.org/files/pervasive-and-persistent-understandings-01-14.pdf'>Pervasive and Persistent Understandings about Data</a> (2014); Data source:<a href='https://www.kaggle.com/dorbicycle/world-foodfeed-production'> Worldwide food\feed production and distribution</a>

## Authentic learning approach

- Learning design based on interdisciplinary<br>
connections and real-world examples
- Industry-aligned data science analysis process
- Python, an all-purpose programming language
- Jupyter notebook, a free industry-standard tool for data scientists
- CallystoHub, free cloud computing

## Learning design and athlete development

### U15 training to train
- Promotes tactical strategies for in-game decision making, reading the situation and inferring 
- Focuses on the team and the process
- Situates personal goals within a team approach

## Learning design and U15 sports analytics

Online communication,<br>
sometimes through shared video analysis spaces

Video replay during games and training

Post–game video analysis, limitted statistics

## Learning design and flexibility
<br>
<img src='./images/flexibility.png' alt='adapting' width='90%' />

## Two data examples

1. Import a csv file and use a Python spreadsheet<br>to create shot maps (ice hockey)
2. Gather data from video to analyze and make decisions (field hockey)

## Data example 1:
## Using IIHF World Junior Championship data to create graphs and a shot map

## Defining ice hockey net zones:<br> What factors can lead to scoring?
<!--USA Hockey Goaltender Basics https://www.usahockeygoaltending.com/page/show/890039-stance-->
||
|-|-|
|<img src='./images/hockey_net_zones.png' width='100%'/>|<img src='https://cdn.hockeycanada.ca/hockey-canada/Team-Canada/Men/Under-18/2014-15/2014-15_goalie_camp.jpg?q=60' />|
||<a href='https://www.hockeycanada.ca/en-ca/news/34-goaltenders-invited-to-2014-poe-camp'>Image source: Hockey Canada</a>|

In [4]:
%%html 
<h2>Data source: Shot charts</h2><br>
<iframe width="850" height="600" src="https://www.iihf.com/pdf/503/ihm503a13_77a_3_0" frameborder="0" ></iframe>

## Generating a csv file

Net_Zone,Austria,Canada,Czech_Republic,Finland,Germany,Russia,Switzerland,Slovakia,Sweden,USA,Total<br>
1,0,7,0,3,2,0,0,0,3,3,18<br>
2,0,1,1,0,1,0,0,0,0,0,3<br>
3,0,5,0,2,2,4,1,0,3,6,23<br>
4,0,4,3,2,1,1,0,1,0,3,15<br>
5,0,1,0,2,1,0,0,0,0,0,4<br>
6,1,1,2,4,0,2,0,1,0,2,13<br>
7,0,6,0,1,3,3,1,1,0,9,24<br>
8,0,5,1,2,2,3,1,2,3,2,21<br>
9,0,3,3,0,2,3,2,0,5,0,18<br>

## Exploring scoring on net zones 

In [5]:
hockey_goals_df = pd.read_csv('./data/hockey_goals.csv')
hockey_goals_df.head(9)

Zone  Austria  Canada  Czech_Republic  Finland  Germany  Russia  \
0    one        0       7               0        3        2       0   
1    two        0       1               1        0        1       0   
2  three        0       5               0        2        2       4   
3   four        0       4               3        2        1       1   
4   five        0       1               0        2        1       0   
5    six        1       1               2        4        0       2   
6  seven        0       6               0        1        3       3   
7  eight        0       5               1        2        2       3   
8   nine        0       3               3        0        2       3   

   Switzerland  Slovakia  Sweden  USA  Total  
0            0         0       3    3     18  
1            0         0       0    0      3  
2            1         0       3    6     23  
3            0         1       0    3     15  
4            0         0       0    0      4  
5            0         1       0    2     13  
6            1         1       0    9     24  
7            1         2       3    2     21  
8            2         0       5    0     18

### What do measures of central tendency<br>tell us about the total goals per net zone?

In [6]:
hockey_goals_df['Total'].sum()

139

In [7]:
hockey_goals_df['Total'].min()

3

In [8]:
hockey_goals_df['Total'].max()

24

In [9]:
hockey_goals_df['Total'].mean()

15.444444444444445

In [10]:
hockey_goals_df['Total'].median()

18.0

In [11]:
hockey_goals_df['Total'].mode()

0    18
dtype: int64

### Which net zones score above the median? 

In [12]:
hockey_goals_df['Total'].median()

18.0

In [13]:
hockey_goals_df.sort_values('Total', ascending=False)

Zone  Austria  Canada  Czech_Republic  Finland  Germany  Russia  \
6  seven        0       6               0        1        3       3   
2  three        0       5               0        2        2       4   
7  eight        0       5               1        2        2       3   
0    one        0       7               0        3        2       0   
8   nine        0       3               3        0        2       3   
3   four        0       4               3        2        1       1   
5    six        1       1               2        4        0       2   
4   five        0       1               0        2        1       0   
1    two        0       1               1        0        1       0   

   Switzerland  Slovakia  Sweden  USA  Total  
6            1         1       0    9     24  
2            1         0       3    6     23  
7            1         2       3    2     21  
0            0         0       3    3     18  
8            2         0       5    0     18  
3            0         1       0    3     15  
5            0         1       0    2     13  
4            0         0       0    0      4  
1            0         0       0    0      3

In [34]:
bar_hockey_goals_df = px.bar(hockey_goals_df,
           x="Zone", y="Total", title="Goals by net zone")

bar_hockey_goals_df.update_layout(title_text='Total goals by net zone')

### What connections exist between<br>goalie position and scoring?

In [15]:
hockey_goals_df = pd.read_csv('./data/hockey_goals.csv') 
hockey_goals_df.Total

0    18
1     3
2    23
3    15
4     4
5    13
6    24
7    21
8    18
Name: Total, dtype: int64

In [16]:
map_hockey_goals_df = px.imshow([[18, 3, 23],[15, 4, 13],[24, 21, 18]])
temp = dict(showticklabels = False)
map_hockey_goals_df.update_layout(xaxis = temp,yaxis = temp,title="2021 WJC goals by net zone")
map_hockey_goals_df.update_layout(autosize=False, width=600, height=400)

### Exploring Canada's scoring on net zones 

In [17]:
spread_sheet_hockey_net = sheet(rows=3, columns=3)

my_cells_net = cell_range([[7,1,5],[4,1,1],[6,5,3]],row_start=0,col_start=0,numeric_format="int")

figure_hockey_net = go.Figure(data=go.Heatmap(
          z =list(reversed(my_cells_net.value)),
          type = 'heatmap',
          colorscale = 'peach',opacity = 1.0))

axis_template = dict(range = [0,5], autorange = True,
             showgrid = False, zeroline = False,
             showticklabels = False,
             ticks = '' )

figure_hockey_net.update_layout(margin = dict(t=50,r=200,b=200,l=200),
    xaxis = axis_template,
    yaxis = axis_template,
    showlegend = False,
    width = 800, height = 500, title="Ice hockey net zones",
    autosize = True )

# Add image in the background
nLanes = 3
nZones = 3

figure_hockey_net.add_layout_image(
        dict(
            source="images/hockey_net_zones.png",
            xref="x",
            yref="y",
            x=-0.5,
            y=-.5 + nLanes,  #this adjusts the placement of the image
            sizex=nZones,
            sizey=nLanes,
            sizing="fill",
            opacity=0.7,
            layer="above")
)

# changes in my_cells should trigger this function
def calculate(change):
    figure_hockey_net.update_traces(z=list(reversed(my_cells_net.value)))
    
my_cells_net.observe(calculate, 'value')

In [18]:
spread_sheet_hockey_net

Sheet(cells=(Cell(column_end=2, column_start=0, numeric_format='int', row_end=2, row_start=0, squeeze_column=F…

In [19]:
figure_hockey_net.update()  # Click the keys "Shift-return" to update the figure

## Data example 2:

## Analyzing field hockey data to make decisions

<center><img src='./images/learning_cycle1.png' alt="Learning design and context" width='90%' /></center>

<center><img src='./images/learning_cycle5.png' alt="Learning cycle" width='90%' /></center>

<center><img src='./images/computational_thinking.png' alt="Computationl thinking" width='90%' /></center>

<center><img src='./images/analysis_process.png' alt="Data science analysis process" width='90%' /></center>

<center><img src='./images/collective.png' alt="Collective decision making" width='90%' /></center>

<center><img src='./images/field_hockey_game.png' alt="Field hockey" width='90%' /></center>

<center><img src='./images/understand1.png' alt="Understand actions" width='90%' /></center>

<center><img src='./images/actions.png' alt="Understand viewpoints" width='90%' /></center>

In [20]:
print ('Passes received')
YouTubeVideo('mIwiiJO7Rk4?start=2893&end=2910', width='600', height='355')

Passes received


In [21]:
print ('Pass intercepted')
YouTubeVideo('mIwiiJO7Rk4?start=968&end=979', width='600', height='355')

Pass intercepted


<center><img src='./images/gather4.png' alt="Gather" width='90%' /></center>

<center><img src='./images/collection_passing.png' alt="Passing" width='90%' /></center>

## 3. Organize

In [22]:
possession_time_df = read_csv('data/field_hockey_possession_time.csv')
possession_time_df.head(8)

Team Quarter  Possession Time (seconds)
0  home   first                        365
1  home  second                        355
2  home   third                        410
3  home  fourth                        390
4  away   first                        535
5  away  second                        545
6  away   third                        490
7  away  fourth                        510

## 4. Explore: How can the data be analyzed to represent viewpoints?

Game summary:

Final score: Home 0, Away 1<br><br>
Home dominated shots on net with 2 shots in the third quarter<br>while Away shot once on net in the first quarter and scored.

**Attack viewpoint: How does ball possession<br>time affect the outcomes of the game?**

In [37]:
bar_possession_time_df = px.bar(possession_time_df,x="Possession Time (seconds)",y="Quarter",title="Possession per quarter",color="Team")
bar_possession_time_df.update_layout(autosize=False, width=600, height=400)

**Attack and defensive transition viewpoints:<br>How does passing within lanes change and stay the same?**

In [24]:
lanes_home_passes_df = read_csv('data/field_hockey_lanes_home_passes.csv')
lanes_home_passes_df.head()

Team         Action Quarter         Lane  Count
0  home  pass received   first   left outer     14
1  home  pass received   first   left inner     10
2  home  pass received   first        inner      5
3  home  pass received   first  right inner      7
4  home  pass received   first  right outer     13

In [25]:
lanes_home_passes_df[lanes_home_passes_df['Quarter'].isin(['first', 'second'])].sum()

Team       homehomehomehomehomehomehomehomehomehomehomeho...
Action     pass receivedpass receivedpass receivedpass re...
Quarter    firstfirstfirstfirstfirstsecondsecondsecondsec...
Lane       left outerleft innerinnerright innerright oute...
Count                                                    113
dtype: object

In [45]:
scatter_lanes_home_passes_q12_df = px.scatter(lanes_home_passes_df[lanes_home_passes_df['Quarter'].isin(['first', 'second'])],x="Lane",y="Quarter", size="Count", color="Action", size_max=59,title="Home possession: Passes per lane for first and second quarters (upfield attack)")
scatter_lanes_home_passes_q12_df.update_layout(autosize=False, width=900)

- For the first half, Home receives the most passes on the outer lanes, especially the left outer lane. 
- During the first quarter home is able to pass into the inner lanes more frequently.
- While Away intercepts Home in the inner lanes 7 times, Home successfully receives 35 passes.

In [27]:
lanes_home_passes_df[lanes_home_passes_df['Quarter'].isin(['third', 'fourth'])].sum()

Team       homehomehomehomehomehomehomehomehomehomehomeho...
Action     pass receivedpass receivedpass receivedpass re...
Quarter    thirdthirdthirdthirdthirdfourthfourthfourthfou...
Lane       left outerleft innerinnerright innerright oute...
Count                                                     88
dtype: object

In [43]:
scatter_lanes_home_passes_q34_df = px.scatter(lanes_home_passes_df[lanes_home_passes_df['Quarter'].isin(['third', 'fourth'])],x="Lane", y="Quarter", size="Count", size_max=59,color="Action", title="Home possession: Passes per lane for third and fourth quarters (downfield attack)")
scatter_lanes_home_passes_q34_df.update_layout(autosize=False, width=900)

- During the second half, Home receives the most passes (n=16) on the right outer lane, which generally is the opponent's strong side. 
- While away intercepts Home in the inner lanes 8 times, Home successfully receives 23 passes.

## 5. Interpret: What connections exist<br> between possession and passing?<br>

|First half|Second half|
|:-|:-|
|Possession time, 720 seconds|Possession time, 800 seconds|
|Total passes 113|Total passes 88|
|Inner lanes passes received 35<br>(40% of total passes)|Inner lanes passes received 23<br>(26% of total passes)|
|7 intercepts in inner lanes|8 intercepts in inner lanes|

>Home shot on net twice during the third quarter,<br> their highest ball possession quarter.
>
>During the second half, Home had more ball possession but received fewer passes.<br>
>This indicates that Home carried the ball more second half, increasing difficulty for Away to challenge them.
>
>Since carrying occurred, the visualizations of passing do not fully represent actions within the inner lanes.

## 4. Explore
### Home attack as passes received per quarter: 
What stays the same and what changes across zones?

In [30]:
df_passes_home = pd.read_csv('data/field_hockey_home_passes.csv'); df_passes_home

Team         Phase of Play                   Action Quarter  Lane  Zone  \
0    home                attack            pass received   first     1     4   
1    home                attack            pass received   first     2     4   
2    home                attack            pass received   first     3     4   
3    home                attack            pass received   first     4     4   
4    home                attack            pass received   first     5     4   
..    ...                   ...                      ...     ...   ...   ...   
235  home  defensive transition  pass intercepted/fouled  fourth     1     1   
236  home  defensive transition  pass intercepted/fouled  fourth     2     1   
237  home  defensive transition  pass intercepted/fouled  fourth     3     1   
238  home  defensive transition  pass intercepted/fouled  fourth     4     1   
239  home  defensive transition  pass intercepted/fouled  fourth     5     1   

     Count  
0        2  
1        1  
2        0  
3        0  
4        1  
..     ...  
235      0  
236      0  
237      2  
238      0  
239      0  

[240 rows x 7 columns]

In [31]:
df_temp_1 = df_passes_home.loc[lambda df: (df['Phase of Play'] == 'attack') &(df['Quarter'] == 'first') ];
df_temp_2 = df_passes_home.loc[lambda df: (df['Phase of Play'] == 'attack') &(df['Quarter'] == 'second') ]; 
df_temp_3 = df_passes_home.loc[lambda df: (df['Phase of Play'] == 'attack') &(df['Quarter'] == 'third') ]; 
df_temp_4 = df_passes_home.loc[lambda df: (df['Phase of Play'] == 'attack') &(df['Quarter'] == 'fourth') ]; 

In [32]:
#Fix: add titles
#import plotly.tools as tls
fig_all = subplots.make_subplots(rows=1, cols=4)
fig_1 = df_temp_1.iplot(kind='heatmap', colorscale='blues', x='Lane', y='Zone', z='Count' , asFigure=True) 
fig_2 = df_temp_2.iplot(kind='heatmap', colorscale='blues', x='Lane', y='Zone', z='Count' , asFigure=True)
fig_3 = df_temp_3.iplot(kind='heatmap', colorscale='blues', x='Lane', y='Zone', z='Count' , asFigure=True)
fig_4 = df_temp_4.iplot(kind='heatmap', colorscale='blues', x='Lane', y='Zone', z='Count' , asFigure=True)
fig_all.append_trace(fig_1['data'][0], 1, 1)
fig_all.append_trace(fig_2['data'][0], 1, 2)
fig_all.append_trace(fig_3['data'][0], 1, 3)
fig_all.append_trace(fig_4['data'][0], 1, 4)

In [46]:
fig_all.update_xaxes(showticklabels = False, linecolor='black')
fig_all.update_yaxes(showticklabels = False, linecolor='black')
iplot(fig_all)

- The most passes in a zone occur toward the left outside lane of the opponent's net across quarters.
- Home starts to apply presssure in the second quarter keeping passes away from the net and on the outer lanes.
- In the second half, defence keeps passes out of the zone of the opponent's net.

## 5. Interpret<br> How can the data exploration inform decision making?

> - keeping ball possession by carrying the ball
> - dominating play by keeping the ball out of the zone near the opponent's net
> - attacking on the outer lanes, especially toward the left side of the opponent's net

<center><a href='https://www.callysto.ca/learning-modules/'><img src='./images/learning_modules.png' target='_blank' alt="Callysto learning modules" width='90%' /></a></center>
<center>All free, all open source, aimed at teachers and students</center>

## Callysto resources

- <a href="https://app.lucidchart.com/documents/view/8e3186f7-bdfe-46af-9c7f-9c426b80d083">Connecting data literacy and sports</a>
- <a href="https://www.callysto.ca/starter-kit/">Callysto starter kit</a>
- <a href="https://courses.callysto.ca">courses.callysto.ca</a>
- <a href="https://www.callysto.ca/weekly-data-visualization/">Weekly data visualizations</a>

<center><h2><a href="https://us19.list-manage.com/survey?u=3a7de3527a6c40f3824010da7&id=d81b0860cf">Feedback form</a></h2></center>

<br>
<p><center>Contact us at <a href="mailto:contact@callysto.ca">contact@callysto.ca</a><br>
for in-class workshops, virtual hackathons...<br>
    <a href="https://twitter.com/callysto_canada">@callysto_canada</a><br>
    <a href="https://callysto.ca">callysto.ca</a><br>
    <a href="https://www.youtube.com/channel/UCPdq1SYKA42EZBvUlNQUAng">YouTube</a>
</center></p>

<center><img src='./images/callysto_logo.png' alt="Callysto logo" width='80%' /></center>
<center><img src='./images/callysto_partners2.png' alt='Callysto partners' width='80%' /></center>

Alberta Education. (2017). *Career and Technology Foundations* [Program of Studies]. https://education.alberta.ca/media/3795641/ctf-program-of-studies-jan-4-2019.pdf

Alberta Education. (2007, updated 2016). *Mathematics Kindergarten to Grade 9* [Program of Studies]. https://education.alberta.ca/media/3115252/2016_k_to_9_math_pos.pdf

Alberta Education. (2000). *Physical Education* [Program of Studies]. https://education.alberta.ca/media/160191/phys2000.pdf

BC Ministry of Education. (2020). *BC's Digital Literacy Framework*. https://www2.gov.bc.ca/assets/gov/education/kindergarten-to-grade-12/teach/teaching-tools/digital-literacy-framework.pdf

Berikan, B., & Özdemir, S. (2020). *Investigating “Problem-Solving With Datasets” as an Implementation of Computational Thinking: A Literature Review*. Journal of Educational Computing Research, 58(2), 502–534. https://doi.org/10.1177/0735633119845694

Borromeo Ferri, R. (2006). *Theoretical and empirical differentiations of phases in the modeling process*. Zentralblatt fuer Didaktik der Mathematik, 38(2), 86-95. ZDM. 38. 86-95. 10.1007/BF02655883. 

Butler, D. & Winne, P. (1995). *Feedback and Self-Regulated Learning: A Theoretical Synthesis*. Review of Educational Research - REV EDUC RES. 65. 245-281. 10.2307/1170684. 

Cleary, T. & Zimmerman, B. (2001). *Self-Regulation Differences during Athletic Practice by Experts, Non-Experts, and Novices*. Journal of Applied Sport Psychology. 13. 185-206. 10.1080/104132001753149883. 

Conseil supérieur de l’éducation (2020). Éduquer au numérique, Rapport sur l’état et les besoins de
l’éducation 2018-2020, Québec, Le Conseil, 96 p.

Field Hockey Alberta. (2020). *Tactical Seminars*. http://www.fieldhockey.ab.ca/content/tactical-seminars

Field Hockey Canada. (2020). *Ahead of the Game*. http://www.fieldhockey.ca/ahead-of-the-game-field-hockey-canada-webinar-series/

Gadanidis, G. (2020, September 2). *Shifting from Computational Thinking to Computational Modelling in Math Education* [Online plenary talk]. Changing the Culture 2020, Pacific Institute for the Mathematical Sciences.

Guadalupe, T. & Gómez-Blancarte, A. (2019). *Assessment of Informal and Formal Inferential Reasoning: A Critical Research Review*. Statistics Education Research Journal, 18, 8-25. https://www.researchgate.net/publication/335057564_ASSESSMENT_OF_INFORMAL_AND_FORMAL_INFERENTIAL_REASONING_A_CRITICAL_RESEARCH_REVIEW

Hadwin, A. F., Järvelä., S., & Miller, M. (2018). Self-regulation, co-regulation, and shared regulation in collaborative learning environments. In D. H. Schunk, & J. A. Greene, (Eds.). *Handbook of self-regulation of learning and performance* (2nd ed.). New York: Routledge. https://www.researchgate.net/publication/313369294_Self-regulation_co-regulation_and_shared_regulation_in_collaborative_learning_environments


Kastens, K. (2014). *Pervasive and Persistent Understandings about Data*. Oceans of Data Institute. http://oceansofdata.org/sites/oceansofdata.org/files/pervasive-and-persistent-understandings-01-14.pdf

Leard, T., & Hadwin, A. F. (2001, May). *Analyzing logfile data to produce Navigation Profiles of Studying as Self-regulated learning*. Paper presented at the annual meeting of the Canadian Society for the Study of Education, Quebec City, Quebec, Canada.

Manitoba Education and Training (2020). *Literacy with ICT across the Curriculum: A Model for 21st Century Learning from K-12*. https://www.edu.gov.mb.ca/k12/tech/lict/index.html

Ontario Ministry of Education. (2020). *The Ontario curriculum grades 1‐8: Mathematics* [Program of Studies]. https://www.dcp.edu.gov.on.ca/en/curriculum/elementary-mathematics